### PR TITLE
feat(v3proxy): improved error handling

### PR DESCRIPTION
**Pull request by kschelonka**
Changes:

* Delegate error handling to middleware methods instead of try/catch on every method
* Only log relevant errors to Sentry/Cloudwatch - bad input, unexpected server errors, etc. Do not log errors returned by the GraphQL server unless they indicate an issue with the proxy code (e.g. bad input).
* Customize error response depending on the error source (validator, graphql client, unexpected/app errors)

Related but unrelated bugfixes:

* Ensure the validator method converts strings to integers where relevant with a custom sanitizer
* Remove an incorrect slash from one of the urls in the graph client initializer

### CODE REVIEW

File: servers/v3-proxy-api/src/graph/graphQLClient.ts [hunk 1]
```diff
@@ -47,22 +46,15 @@ export function getClient(
...
-    serverLogger.error(e);
-    throw e;
+    url = `${config.graphQLProxy}?consumer_key=${consumerKey}`;
   }
+  Sentry.addBreadcrumb({ message: `creating GraphQL client` });
+  return new GraphQLClient(url, {
+    headers: headers,
+    //fetch implementation used by node version,
+    //can give custom fetch package
+    fetch,
```
Comment (bassrock) [inline servers/v3-proxy-api/src/graph/graphQLClient.ts:56]: ooo thats right node includes this now!

File: servers/v3-proxy-api/src/middleware/errorHandlers.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ */
+export function customErrorHeaders(errorCode: unknown, language = 'en') {
+  if (
+    typeof errorCode !== 'string' ||
+    errorCode == null ||
+    errorHeaders[errorCode] == null
+  ) {
+    return undefined;
+  }
+  //todo: proxy should handle localization based on web repo request.
```
Comment (bassrock) [inline servers/v3-proxy-api/src/middleware/errorHandlers.ts:38]: For the time being i think returning only english to devs for now is fine

**Review by bassrock — APPROVED** [review]
I really like the error abstraction patterns here, This will make things a lot easier.

File: servers/v3-proxy-api/src/middleware/errorHandlers.ts (hunk 2 above)
Comment (kschelonka) [reply servers/v3-proxy-api/src/middleware/errorHandlers.ts:38]: github tricks - this is an old comment, just unindented I think?

File: servers/v3-proxy-api/src/graph/graphQLClient.ts (hunk 1 above)
Comment (kschelonka) [reply servers/v3-proxy-api/src/graph/graphQLClient.ts:56]: can't take credit, this was just unindented I think haha

**Pull request merged**